test/controllers/benefits/CompanyBenefitControllerSpec.scala

```diff
@@ -194,6 +210,64 @@ class CompanyBenefitControllerSpec
       }
     }
 
+    "redirect to the Tax Account Summary Page (start of journey)" when {
+      "the form has the value noIDontGetThisBenefit and EndCompanyBenefit_BenefitTypeKey is not cached" in {
+        val SUT = createSUT
+        when(
+          journeyCacheService.mandatoryJourneyValueAs(eqTo(EndCompanyBenefit_BenefitTypeKey), any())(
+            any()))
+          .thenReturn(Future.successful(Left("")))
+
+        val result = SUT.submitDecision(
+          RequestBuilder
+            .buildFakeRequestWithAuth("POST")
+            .withFormUrlEncodedBody(DecisionChoice -> NoIDontGetThisBenefit))
+
+        val redirectUrl = redirectLocation(result)
+
+        redirectUrl mustBe Some(controllers.routes.TaxAccountSummaryController.onPageLoad().url)
+
+      }
+
+      "the form has the value YesIGetThisBenefit and EndCompanyBenefit_BenefitTypeKey is not cached" in {
+        val SUT = createSUT
+        when(
+          journeyCacheService.mandatoryJourneyValueAs(eqTo(EndCompanyBenefit_BenefitTypeKey), any())(
+            any()))
+          .thenReturn(Future.successful(Left("")))
+
+        val result = SUT.submitDecision(
+          RequestBuilder
+            .buildFakeRequestWithAuth("POST")
+            .withFormUrlEncodedBody(DecisionChoice -> YesIGetThisBenefit))
+
+        val redirectUrl = redirectLocation(result).getOrElse("")
+
+        redirectUrl mustBe controllers.routes.TaxAccountSummaryController.onPageLoad()
+
+      }
+
+      "the form has no valid value" in{
+        val SUT = createSUT
+        val benefitType = "Telephone"
+        when(
+          journeyCacheService.mandatoryJourneyValueAs[String](eqTo(EndCompanyBenefit_BenefitTypeKey), any())(
+            any()))
+          .thenReturn(Future.successful(Right(benefitType)))
+
+        val result = SUT.submitDecision(
+          RequestBuilder
+            .buildFakeRequestWithAuth("POST")
+            .withFormUrlEncodedBody(DecisionChoice -> Random.alphanumeric.take(10).mkString))
+
+        status(result) mustBe SEE_OTHER
+
+        val redirectUrl = redirectLocation(result).getOrElse("")
+
+        redirectUrl mustBe controllers.routes.TaxAccountSummaryController.onPageLoad().url
+      }
+    }
```

Comment:
Sounds overkill but there should be a test for the actual bug of where there is a good (Right result from mandatoryJourneyValueAs) but it does not match with the user's current journey

Reply:
As in, returns Yes when it should be No? Or do you mean it returns Apples when it should be Yes or No?

Reply:
As in, the cache is set up to return one benefit and when you start the journey on another benefit it's not prepopulated - basically a test for the scenario described in the ticket. If it's not possible here then it needs to be done in an acceptance test.

Reply:
I'm going to handle this with an acceptance test.